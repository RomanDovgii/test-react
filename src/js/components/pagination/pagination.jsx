import React from "react";

const Pagination = () => {
  return (
    <ul className="pagination">
      <li className="pagination__item"><a href="#" className="pagination__link">&lt;&lt;</a></li>
      <li className="pagination__item"><a href="#" className="pagination__link">&lt;</a></li>
      <li className="pagination__item"><a href="#" className="pagination__link">1</a></li>
      <li className="pagination__item"><a href="#" className="pagination__link">2</a></li>
      <li className="pagination__item"><a href="#" className="pagination__link">3</a></li>
      <li className="pagination__item"><a href="#" className="pagination__link">4</a></li>
      <li className="pagination__item"><a href="#" className="pagination__link">5</a></li>
      <li className="pagination__item"><a href="#" className="pagination__link">...</a></li>
      <li className="pagination__item"><a href="#" className="pagination__link">56</a></li>
      <li className="pagination__item"><a href="#" className="pagination__link">57</a></li>
      <li className="pagination__item"><a href="#" className="pagination__link">58</a></li>
      <li className="pagination__item"><a href="#" className="pagination__link">59</a></li>
      <li className="pagination__item"><a href="#" className="pagination__link">60</a></li>
      <li className="pagination__item"><a href="#" className="pagination__link">&gt;</a></li>
      <li className="pagination__item"><a href="#" className="pagination__link">&gt;&gt;</a></li>
    </ul>
  );
};

export default Pagination;
