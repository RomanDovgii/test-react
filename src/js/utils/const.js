export const MAXIMUM_USERS_PER_PAGE = 50;

export const AppRoute = {
  SELECT: `/select`,
  TABLE: `/table`
};
