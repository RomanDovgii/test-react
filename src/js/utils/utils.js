export const extend = (a, b) => {
  return Object.assign({}, a, b);
};

export const sortToTop = (a, b) => {
  switch (true) {
    case a > b:
      return -1;
    case a < b:
      return 1;
    default:
      return 0;
  }
};

export const sortToBot = (a, b) => {
  switch (true) {
    case a < b:
      return -1;
    case a > b:
      return 1;
    default:
      return 0;
  }
};
