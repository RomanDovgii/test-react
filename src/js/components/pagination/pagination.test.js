import React from "react";
import ShallowRenderer from "react-test-renderer/shallow";
import {Pagination} from "./pagination";
import {noop} from "../../utils/utils";

it(
    `Shouldn't render Pagination`,
    () => {
      const renderer = new ShallowRenderer();

      const tree = renderer
      .render(
          <Pagination
            pages={1}
            currentPage={1}
            onPageClick={noop}
            onFirstClick={noop}
            onPreviousClick={noop}
            onNextClick={noop}
            onLastClick={noop}
          />
      );

      expect(tree).toMatchSnapshot();
    }
);

it(
    `Should render correct Pagination with 5 pages`,
    () => {
      const renderer = new ShallowRenderer();

      const tree = renderer
      .render(
          <Pagination
            pages={5}
            currentPage={1}
            onPageClick={noop}
            onFirstClick={noop}
            onPreviousClick={noop}
            onNextClick={noop}
            onLastClick={noop}
          />
      );

      expect(tree).toMatchSnapshot();
    }
);

it(
    `Should render correct Pagination with 10 pages`,
    () => {
      const renderer = new ShallowRenderer();

      const tree = renderer
      .render(
          <Pagination
            pages={10}
            currentPage={1}
            onPageClick={noop}
            onFirstClick={noop}
            onPreviousClick={noop}
            onNextClick={noop}
            onLastClick={noop}
          />
      );

      expect(tree).toMatchSnapshot();
    }
);

it(
    `Should render correct Pagination with 10 pages, user should be on fifth page`,
    () => {
      const renderer = new ShallowRenderer();

      const tree = renderer
      .render(
          <Pagination
            pages={10}
            currentPage={5}
            onPageClick={noop}
            onFirstClick={noop}
            onPreviousClick={noop}
            onNextClick={noop}
            onLastClick={noop}
          />
      );

      expect(tree).toMatchSnapshot();
    }
);
