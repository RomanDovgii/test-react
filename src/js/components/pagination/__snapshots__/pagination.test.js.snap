// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render correct Pagination with 5 pages 1`] = `
<ul
  className="pagination"
>
  <li
    className="pagination__item"
  >
    <a
      className="pagination__link pagination__link--active"
      href="#"
      onClick={[Function]}
    >
      1
    </a>
  </li>
  <li
    className="pagination__item"
  >
    <a
      className="pagination__link "
      href="#"
      onClick={[Function]}
    >
      2
    </a>
  </li>
  <li
    className="pagination__item"
  >
    <a
      className="pagination__link "
      href="#"
      onClick={[Function]}
    >
      3
    </a>
  </li>
  <li
    className="pagination__item"
  >
    <a
      className="pagination__link "
      href="#"
      onClick={[Function]}
    >
      4
    </a>
  </li>
  <li
    className="pagination__item"
  >
    <a
      className="pagination__link "
      href="#"
      onClick={[Function]}
    >
      5
    </a>
  </li>
  <React.Fragment>
    <li
      className="pagination__item"
    >
      <a
        className="pagination__link"
        href="#"
        onClick={[Function]}
      >
        &gt;
      </a>
    </li>
    <li
      className="pagination__item"
    >
      <a
        className="pagination__link"
        href="#"
        onClick={[Function]}
      >
        &gt;&gt;
      </a>
    </li>
  </React.Fragment>
</ul>
`;

exports[`Should render correct Pagination with 10 pages 1`] = `
<ul
  className="pagination"
>
  <li
    className="pagination__item"
  >
    <a
      className="pagination__link pagination__link--active"
      href="#"
      onClick={[Function]}
    >
      1
    </a>
  </li>
  <li
    className="pagination__item"
  >
    <a
      className="pagination__link "
      href="#"
      onClick={[Function]}
    >
      2
    </a>
  </li>
  <li
    className="pagination__item"
  >
    <a
      className="pagination__link "
      href="#"
      onClick={[Function]}
    >
      3
    </a>
  </li>
  <li
    className="pagination__item"
  >
    <a
      className="pagination__link "
      href="#"
      onClick={[Function]}
    >
      4
    </a>
  </li>
  <li
    className="pagination__item"
  >
    <a
      className="pagination__link "
      href="#"
      onClick={[Function]}
    >
      5
    </a>
  </li>
  <li
    className="pagination__item"
  >
    <a
      className="pagination__link "
      href="#"
      onClick={[Function]}
    >
      6
    </a>
  </li>
  <li
    className="pagination__item"
  >
    <a
      className="pagination__link "
      href="#"
      onClick={[Function]}
    >
      7
    </a>
  </li>
  <li
    className="pagination__item"
  >
    <a
      className="pagination__link "
      href="#"
      onClick={[Function]}
    >
      8
    </a>
  </li>
  <li
    className="pagination__item"
  >
    <a
      className="pagination__link "
      href="#"
      onClick={[Function]}
    >
      9
    </a>
  </li>
  <li
    className="pagination__item"
  >
    <a
      className="pagination__link "
      href="#"
      onClick={[Function]}
    >
      10
    </a>
  </li>
  <React.Fragment>
    <li
      className="pagination__item"
    >
      <a
        className="pagination__link"
        href="#"
        onClick={[Function]}
      >
        &gt;
      </a>
    </li>
    <li
      className="pagination__item"
    >
      <a
        className="pagination__link"
        href="#"
        onClick={[Function]}
      >
        &gt;&gt;
      </a>
    </li>
  </React.Fragment>
</ul>
`;

exports[`Should render correct Pagination with 10 pages, user should be on fifth page 1`] = `
<ul
  className="pagination"
>
  <React.Fragment>
    <li
      className="pagination__item"
    >
      <a
        className="pagination__link"
        href="#"
        onClick={[Function]}
      >
        &lt;&lt;
      </a>
    </li>
    <li
      className="pagination__item"
    >
      <a
        className="pagination__link"
        href="#"
        onClick={[Function]}
      >
        &lt;
      </a>
    </li>
  </React.Fragment>
  <li
    className="pagination__item"
  >
    <a
      className="pagination__link "
      href="#"
      onClick={[Function]}
    >
      1
    </a>
  </li>
  <li
    className="pagination__item"
  >
    <a
      className="pagination__link "
      href="#"
      onClick={[Function]}
    >
      2
    </a>
  </li>
  <li
    className="pagination__item"
  >
    <a
      className="pagination__link "
      href="#"
      onClick={[Function]}
    >
      3
    </a>
  </li>
  <li
    className="pagination__item"
  >
    <a
      className="pagination__link "
      href="#"
      onClick={[Function]}
    >
      4
    </a>
  </li>
  <li
    className="pagination__item"
  >
    <a
      className="pagination__link pagination__link--active"
      href="#"
      onClick={[Function]}
    >
      5
    </a>
  </li>
  <li
    className="pagination__item"
  >
    <a
      className="pagination__link "
      href="#"
      onClick={[Function]}
    >
      6
    </a>
  </li>
  <li
    className="pagination__item"
  >
    <a
      className="pagination__link "
      href="#"
      onClick={[Function]}
    >
      7
    </a>
  </li>
  <li
    className="pagination__item"
  >
    <a
      className="pagination__link "
      href="#"
      onClick={[Function]}
    >
      8
    </a>
  </li>
  <li
    className="pagination__item"
  >
    <a
      className="pagination__link "
      href="#"
      onClick={[Function]}
    >
      9
    </a>
  </li>
  <li
    className="pagination__item"
  >
    <a
      className="pagination__link "
      href="#"
      onClick={[Function]}
    >
      10
    </a>
  </li>
  <React.Fragment>
    <li
      className="pagination__item"
    >
      <a
        className="pagination__link"
        href="#"
        onClick={[Function]}
      >
        &gt;
      </a>
    </li>
    <li
      className="pagination__item"
    >
      <a
        className="pagination__link"
        href="#"
        onClick={[Function]}
      >
        &gt;&gt;
      </a>
    </li>
  </React.Fragment>
</ul>
`;

exports[`Shouldn't render Pagination 1`] = `null`;
